src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -604,6 +606,38 @@ public boolean attackEntityFrom(@NotNull final DamageSource damageSource, final
     }
 
     /**
+     * Trigger the corresponding death achievement.
+     * @param source    The damage source.
+     * @param job       The job of the citizen.
+     */
+    public void triggerDeathAchievement(final DamageSource source, final AbstractJob job)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "triggerDeathAchievement" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)